app/controllers/hostgroups_controller.rb

```diff
@@ -102,8 +103,31 @@ def process_hostgroup
     render :partial => "form"
   end
 
+  def taxonomy_warnings
+    @warnings ||= []
+
+    @warnings += organization_warnings(@hostgroup)
+    @warnings += location_warnings(@hostgroup)
+  end
+
   private
 
+  def organization_warnings(hostgroup)
+    warnings = []
+
+    organizations_count = @hostgroup.hosts.unscoped.pluck(:organization_id).uniq.count
```

Comment:
if you replace the `pluck` with `select`, the calculation will be done in DB instead of in memory.